maintenance/log/sink.go

```diff
@@ -29,6 +29,18 @@ func SinkFromContext(ctx context.Context) (*Sink, bool) {
 	return sink, ok
 }
 
+// ContextTransfer gets the logger and sink from the sourceCtx
+// and returns a new context based on targetCtx and these two.
+// If there is no sink in sourceCtx, a new one is created.
+func ContextTransfer(sourceCtx, targetCtx context.Context) context.Context {
+	sink, ok := SinkFromContext(sourceCtx)
+	if !ok {
+		sink = new(Sink)
+	}
+
+	return ContextWithSink(targetCtx, sink)
+}
+
```

Comment:
It doesn't actually transfer the logger. It only transfers the sink. It should transfer both.

Reply:
It does in `ContextWithSInk` but doesn't matter I changed it.